module-4/backend/week_two.md

```diff
@@ -1,22 +1,59 @@
+ES6 - backticks and ${}
+
+2. What's the difference between asynchronous and synchronous JavaScript?
+JS is synchronous - it's not a multi-threaded language
+it only appears to be multi-threaded because the browser's api allows for consumption of functions with callbacks - which get passed along back into the status que, making it appear as if it can perform multiple tasks at one time
+AKA: it's a fraud
+
 3. What are the four pillars of Object Oriented programming?
+shahada, sawm, hajj, and... wait, those are the 5 pillars of islam.
+
+Encapsulation - wrapping methods and functions in a class
+Abstraction - creating classes, objects, etc. in order to separate them by functionality and interfaces
+Inheritance - can reuse code of existing objects, which cuts back on redundancy
+Polymorphism - you can use the same interface for different data types (active record assoc. and arrays, for example)
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+You can create 'objects' in JS - with properties, and properties can even be functions on that particular object.  You can create constructor functions, which are basically an object factory.  You can abstract by functionality.
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+functional refactoring *seems* to be preferred? Ajax requests, event listeners, the basic model, and response handlers are separated by their function
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
-7. What are the different scopes of variables in Javascript? When would you want to define global variables?
+a callback function is a function called by another function - we might use one inside of an API call, so that the browser's API takes care of the ajax call, and then the callback function is passed along to re-enter the status que when all the other functions have finished
+
+7. What's the scope of variables in Javascript?
+the scope of a variable refers to the context in which the variable is accessible.  Whether that is global, only within a particular function, due to circumstances, etc
+
 8. What's the difference between `==` and `===` in JavaScript?
+loose equality (==) vs strict equality (===)
+loose equality compares one piece to another, but *can* convert the type of the values to match - which I think we can all agree, is cheating
+strict equality (aka: the best equality) - does not convert either value to 'force' a match
+
 9. Why do front end frameworks exist?
+because JS is s%i*, and was written in like, 10 days.  You need an interpreter/someone who can compress/sometimes completely overhaul the existing libraries, because even those are dubious
 
 #### Review  
 
 10. Why do people say "HTTP is stateless"?
+the connection between the server and the browser ends once the transaction does
+
 11. Describe a RESTful API.
+an api where the endpoints follow restful patterns - verb/uri patterns
```

Comment:
Could use a bit more detail here. This is a good resource https://restfulapi.net/